libs/application/template-api-modules/src/lib/modules/templates/driving-license-submission/driving-license-submission.service.ts

```diff
@@ -137,16 +142,39 @@ export class DrivingLicenseSubmissionService extends BaseTemplateApiService {
     const email = answers.email as string
     const phone = answers.phone as string
 
+    const postHealthDeclaration = async (
+      nationalId: string,
+      answers: FormValue,
+      auth: User,
+    ) => {
+      await this.drivingLicenseService
+        .postHealthDeclaration(
+          nationalId,
+          PostTemporaryLicenseWithHealthDeclarationMapper(
+            answers as DrivingLicenseSchema,
+          ),
+          auth.authorization.split(' ')[1] ?? '',
+        )
+        .catch((e) => {
+          throw new Error(
+            `Unexpected error (creating driver's license with health declarations): '${e}'`,
+          )
+        })
+    }
+
     if (applicationFor === 'B-full') {
       return this.drivingLicenseService.newDrivingLicense(nationalId, {
         jurisdictionId: jurisdictionId as number,
         needsToPresentHealthCertificate: needsHealthCert || remarks,
         needsToPresentQualityPhoto: needsQualityPhoto,
       })
     } else if (applicationFor === 'B-temp') {
+      if (needsHealthCert) {
+        await postHealthDeclaration(nationalId, answers, auth)
+      }
       return this.drivingLicenseService.newTemporaryDrivingLicense(
         nationalId,
-        auth,
+        auth.authorization.split(' ')[1] ?? '',
```

Comment:
For something more robust against refactoring done by other programmers. Maybe you could do
```typescript
auth.authorization.replace('Bearer ', '')
```

Note the space in `Bearer `.